### PR TITLE
Cleanup and followed artists

### PR DESCRIPTION
**Pull request by sweir27**
This PR looks kinda big, but it's mostly just cleanup + 1 new feature (the "Works by artists you follow" component).

Things in this PR:

#### CSS
- doubled padding on left/right sides
- left align both icon + lot number on list view
- truncation on titles/names to make them single rows
image
- make the entire artwork component clickable (for grid and list views)
- update styles on list component to match specs
image
- update styles on "my active bids" component to match specs
image
- Fixed the UI for the "Artists" aggregation to include "Artists You Follow" as a separate choice. This also removes some gravity work I thought I had to do 😄 
image

#### Minor things
- Enable `AddToCalendar` component
- `JumpView` should jump to the top of commercial filtering section
  - In order to do this, I decided to update the `JumpView` component to accept DOM elements & offset to jump to, so their positions can be evaluated when the button is clicked (and not when the component is mounted). This means that if the UI changes, such as a new active bid showing up or whatnot, it will always jump to where we want it to.
jump-view

#### Slightly major things
- Added a "Works by artists you follow" component. This makes a _second_ request to gravity, just for the works by followed artists in the given sale, if you are a logged-in user. **Note** A performance optimization could be progressively fetch the artworks when each new page of the carousel is clicked, but this provides an OK baseline.
works-by-follows

### CODE REVIEW

**Comment by acjay** [comment]
Silly question this late in the game, but what's the function of the gavel icons next to the prices?

**Comment by sweir27** [comment]
It's a paddle :) we use that symbol consistently around artsy to indicate that the artwork is in an auction.

File: desktop/apps/auction2/templates/my_active_bids.jade [hunk 1]
```diff
@@ -2,37 +2,33 @@ include ../../../components/bid_status/index
 
 if myActiveBids && myActiveBids.length
   h2 Your Active Bids
-  table
```
Comment (sweir27) [inline desktop/apps/auction2/templates/my_active_bids.jade]: No significant changes here, just converting from `table` to flexbox resulted in a big diff.

File: desktop/apps/auction2/actions.js [hunk 2]
```diff
@@ -75,11 +88,49 @@ export function fetchArtworks() {
...
+        }
+      })
+      if (filter_sale_artworks.hits.length > 0) {
+        dispatch(updateSaleArtworksByFollowedArtists(filter_sale_artworks.hits))
+        dispatch(updateSaleArtworksByFollowedArtistsTotal(filter_sale_artworks.counts.total))
+        dispatch(updateIsLastFollowedArtistsPage())
+        dispatch(showFollowedArtistsRail())
+      }
+    } catch(error) {
+      console.error('error!', error)
```
Comment (damassi) [inline desktop/apps/auction2/actions.js]: Should we implement some kind of `<Error />` component, or a modal or something for error states?

File: desktop/apps/auction2/client.js [hunk 3]
```diff
@@ -33,7 +32,14 @@ const clock = new ClockView({
...
 clock.start()
 
-if (sd.AUCTION && sd.AUCTION.is_live_open == false) {
+const calendar = new AddToCalendar({
+  el: $('.auction2-callout')
+})
+
+// Render my active bids if a user is present and
+// the auction is open and not in live integration mode
+if (user && sd.AUCTION && sd.AUCTION.is_open && sd.AUCTION.is_live_open == false) {
```
Comment (damassi) [inline desktop/apps/auction2/client.js]: `===`

Comment (sweir27) [inline desktop/apps/auction2/client.js]: Agh. Old habits.

File: desktop/apps/auction2/components/artist_filter/index.js [hunk 4]
```diff
@@ -8,14 +8,34 @@ function ArtistFilter(props) {
...
-    updateArtistParamsAction
+    numArtistsYouFollow,
+    updateArtistParamsAction,
+    updateArtistsYouFollowParamAction,
+    user
   } = props
   const artistIds = filterParams.artist_ids
   const allArtists = { id: 'artists-all', name: 'All' }
-  const allArtistsSelected = artistIds.length == 0
+  const allArtistsSelected = artistIds.length == 0 && !filterParams.include_artworks_by_followed_artists
```
Comment (damassi) [inline desktop/apps/auction2/components/artist_filter/index.js]: `===`

File: desktop/apps/auction2/actions.js (hunk 2 above)
Comment (sweir27) [inline desktop/apps/auction2/actions.js]: Maybe, but these could be errors in any part of the system, so I think for now the fact that it renders 0 artworks + the console indication is okay.

File: desktop/apps/auction2/components/auction_page/index.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import CommercialFilter from '../commercial_filter'
+import WorksByFollowedArtists from '../works_by_followed_artists'
+import { default as React, PropTypes } from 'react';
```
Comment (damassi) [inline desktop/apps/auction2/components/auction_page/index.js]: `import React, { PropTypes } from 'react'` (and elsewhere)

File: desktop/apps/auction2/reducers.js [hunk 6]
```diff
@@ -46,6 +53,14 @@ const initialState = {
 
 function auctionArtworks(state = initialState, action) {
   switch (action.type) {
+    case actions.DECREMENT_FOLLOWED_ARTISTS_PAGE: {
+      const currentPage = state.followedArtistRailPage
+      if (!(currentPage === 1)) {
```
Comment (damassi) [inline desktop/apps/auction2/reducers.js]: Could be misunderstanding, but why not `!== 1`?

**Comment by acjay** [comment]
Ah, my bad.

I guess it's more of a design question for @katarinabatina, but since the paddle icon almost looks like a character and is in-line with the text of the price, is it potentially distracting? It's kind of clashing with the pound sign for me.

I'm noticing that there may be ever-so-slightly more space between the paddle and the currency symbol in the module at the bottom of the artist page in production, which seems to have less of this effect:

image

Or maybe it's just an optical illusion!

Could the paddle have a tap/hover pop-up to explain what it symbolizes?

File: desktop/apps/auction2/actions.js (hunk 2 above)
Comment (damassi) [inline desktop/apps/auction2/actions.js]: 👍

**Comment by damassi** [comment]
@acjay I've been consistently confused by it in the past, too.

**Comment by damassi** [comment]
@sweir27 - only a few super minor comments!

File: desktop/apps/auction2/components/works_by_followed_artists/index.js [hunk 7]
```diff
@@ -0,0 +1,82 @@
+import { nextPageOfFollowedArtistArtworks, previousPageOfFollowedArtistArtworks } from '../../actions'
+import classNames from 'classnames'
+import AuctionGridArtwork from '../auction_grid_artwork'
+import ChevronLeft from '../../../../components/main_layout/public/icons/chevron-left.svg'
+import ChevronRight from '../../../../components/main_layout/public/icons/chevron-right.svg'
+import { default as React, PropTypes } from 'react';
```
Comment (damassi) [inline desktop/apps/auction2/components/works_by_followed_artists/index.js]: `React, { PropTypes } `

**Comment by katarinabatina** [comment]
@acjay this is an icon we've rolled out to all auction artworks on the site to give them distinction in the inventory. This icon is a common used notation by our competitors and I think serves the purpose to call attention to these lots that are only available for purchase for a limited time. The issue I see right now is that the height of the icon is taller than the actual type so it should come down a few sizes, perhaps a touch more space separating it from the currency symbol could help as well. Adding a small hover state to all these works might not scale well to every single canvas on Artsy but it I could explore it!

File: desktop/apps/auction2/test/reducers.js [hunk 8]
```diff
@@ -22,6 +26,80 @@ describe('Reducers', () => {
...
+
+      describe('#fetchArtworksByFollowedArtists', () => {
+        let store
+
+        beforeEach(() => {
+          const middlewares = [ thunk ]
+          const mockStore = configureMockStore(middlewares)
+
+          store = mockStore(initialResponse)
+          ActionsRewireApi.__Rewire__('metaphysics', sinon.stub().returns(Promise.resolve(
```
Comment (damassi) [inline desktop/apps/auction2/test/reducers.js]: 👍

File: desktop/apps/auction2/test/reducers.js [hunk 9]
```diff
@@ -22,6 +26,80 @@ describe('Reducers', () => {
...
+            },
+            {
+              type: 'UPDATE_SALE_ARTWORKS_BY_FOLLOWED_ARTISTS_TOTAL',
+              payload: { total: 123 }
+            },
+            { type: 'UPDATE_IS_LAST_FOLLOWED_ARTISTS_PAGE' },
+            { type: 'SHOW_FOLLOWED_ARTISTS_RAIL' }
+          ]
+          store.dispatch(actions.fetchArtworksByFollowedArtists()).then(() => {
+            store.getActions().should.eql(expectedActions)
```
Comment (damassi) [inline desktop/apps/auction2/test/reducers.js]: Where does `getActions()` come from?

Comment (damassi) [inline desktop/apps/auction2/test/reducers.js]: Oh nice: http://arnaudbenard.com/redux-mock-store/

File: package.json [hunk 10]
```diff
@@ -27,11 +27,13 @@
     "babel-core": "^6.17.0",
     "babel-eslint": "^7.0.0",
     "babel-plugin-inline-react-svg": "^0.2.0",
+    "babel-plugin-rewire": "^1.0.0",
     "babel-plugin-transform-class-properties": "^6.23.0",
     "babel-plugin-transform-runtime": "^6.15.0",
     "babel-preset-es2015": "^6.22.0",
     "babel-preset-react": "^6.22.0",
     "babel-preset-stage-3": "^6.17.0",
+    "babel-rewire-plugin": "^0.1.0",
```
Comment (damassi) [inline package.json]: Do we need both `babel-rewire-plugin` and `babel-plugin-rewire`?

**Comment by acjay** [comment]
@katarinabatina Thanks for the context! Yeah, if there's a good way to connect it to an explanation, the "limited-time" aspect of it seems like the part to highlight. I hear you on hover-states, which don't do so well for touch environments.

Supposing it were a pop-up explanation, and that pop-up could be triggered by tapping, what if the pop-up ended with a "view lot to bid" CTA? So even if the user triggered the pop-up by accident, meaning to bid, they wouldn't have to go backward by dismissing it to get to the lot page.

File: desktop/apps/auction2/reducers.js (hunk 6 above)
Comment (sweir27) [inline desktop/apps/auction2/reducers.js]: 😅 no reason at all

**Comment by katarinabatina** [comment]
@acjay to me that solution presents lot of interface/interaction for information that is available to the user if they simply click through to the lot. I don't think the icon introduces a level of ambiguity that would prevent the user from clicking the work.

**Comment by damassi** [comment]
lgtm 👍

**Pull request merged**